### ISSUE TITLE
clarification on dec_inp 

### ISSUE THREAD

**Issue by aaxwaz** [guillaume-chevalier/seq2seq-signal-prediction#6]
I really like this tutorial, however, I have a question regarding your design of dec_inp: 

I understand you are using a zero vector followed by all _enc_inp_ except last one as your _dec_inp_. 

My question is, why not, say, feed the correct input of time step t (even if you may make a mistake at prediction) as your encoder input at each time step, during training? And during inference, since we do not have ground truth, feed your output from decoder at time t as input to decoder for time t+1, and so on?

**Comment by guillaume-chevalier** [comment]
I think I answered this here long ago, if this can still be of any help: 
https://www.reddit.com/r/MachineLearning/comments/6gt65z/d_does_it_make_sense_to_predictregress_time/diz12mm/ 

TLDR: I think that doing feedback of outputs into dec_inp (decoder inputs) may cause weird feedback loops and diverging predictions, except if during train-time we also do such a feedback loop. Let me know if you have other questions.
